### PR TITLE
Initial design of test cases

### LINKED ISSUE #2: Set up test case functions for API endpoints
Goal: add a second API endpoint `/grade` which does the same as `/predict` but it returns the actual value of G3 temporarily modify the model builder accordingly). Then modify `test_routes.py` so that it has the following test setup:

- Has a function `test_predict_accuracy` which takes in a DataFrame of student data and an accuracy ratio (both provided by fixtures), runs each of them through the `/predict` endpoint, counts how many of them are correct and verifies that this is at least the accuracy ratio.
- Another function `test_grade_accuracy` which does the same thing but with the `/grade` endpoint. It will verify that the average absolute difference between the output and expected value is below a certain threshold (also provided by a fixture).
- Create the `student_data`, `acceptable_predict_accuracy`, and `acceptable_grade_tolerance` fixtures in another file populated with data from the csv used for the model.

Priority: high

Time estimate: 1 day (I am very familiar with pytest and we have already discussed how we want to set up the tests)

### PR DESCRIPTION
**Pull request by knassre-cmu**
Resolves #2 

Makes all the changes described in the issue:

- Modifies model_build.ipynb to output the grade (for now)
- Adds a second API route `/grade` that outputs the grade value itself
- Adds the fixtures and tests to `test_routes.py` as described in the issue

Ancillary fix: had to update the pipfile since newer sklearn versions renamed some modules, causing import errors.

### CODE REVIEW

File: app/tests/test_routes.py [hunk 1]
```diff
@@ -13,3 +51,90 @@ def test_base_route():
...
+       considered sufficiently correct."""
+    return 0.75
+
+@pytest.fixture
+def acceptable_grade_tolerance():
+    """The maximum average error required for the grade predictions to be
+       considered sufficiently correct."""
+    return 2
+    
+# Todo: fill in actual data
```
Comment (knassre-cmu) [inline app/tests/test_routes.py]: Remove this comment

File: app/tests/test_routes.py [hunk 2]
```diff
@@ -13,3 +51,90 @@ def test_base_route():
...
+@pytest.fixture(
+    params=[
+        pytest.param(training_indeces, id="training"),  
+        pytest.param(testing_indeces, id="training"),
+        pytest.param(combined_indeces, id="combined"),
```
Comment (knassre-cmu) [inline app/tests/test_routes.py]: Minor thing, but `indices` is spelled wrong in a couple places

File: app/tests/test_routes.py [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
+import pandas as pd
+import csv
+import math
+import random
 
 from app.handlers.routes import configure_routes
 
+#training_indeces = random.sample(range(0, data_file_len), num_training_students)
+
+#generated only once from this code: random.sample(range(0, data_file_len), num_training_students)
```
Comment (knassre-cmu) [inline app/tests/test_routes.py]: Delete

**Review by knassre-cmu — COMMENTED** [review]
I'll make several of these revisions manually, + fixture-ify this stuff to make the tests easier to read
